automated/lib/sh-test-lib

```diff
@@ -476,3 +476,85 @@ generate_skipfile() {
     cat "${SKIPFILE_PATH}"
     info_msg "EOF"
 }
+
+# Description: parted is not working on SSD (Solid-State Drives) this is the
+# reason to use fdisk instead.
+# Dependency packages for fdisk and mkfs are "fdisk e2fsprogs dosfstools"
+# fdisk and mkfs commands might be in $PATH /sbin or /usr/sbin or
+# /usr/local/sbin export PATH for better usage.
+export PATH=$PATH:/sbin:/usr/sbin:/usr/local/sbin
+
+
+# Usage:
+# partition_disk "/dev/sdb" "+5G +10G"
+# shellcheck disable=SC2039
+partition_disk() {
+    [ "$#" -ne 2 ] && error_msg "Usage: <block-device> <sizes>"
+    local device="$1"
+    local sizes="$2"
+
+    # Create a new empty DOS partition table
+    (
+    echo o
+    echo w) | fdisk "${device}"
+
+    if [ -n "${sizes}" ]; then
+        # Create partitions as per sizes
+        for size in ${sizes}; do
+            # Create patitions with ${size}
+            (
+            echo n
+            echo p
+            echo
+            echo
+            echo "${size}"
+            echo w) | fdisk "${device}"
+            check_return "fdisk-${device}-${size}-partition"
+            # sync and wait 5 sec
+            sync
+            sleep 5
+        done
+    fi
+
+    # Create a partition at the end.
+    # Use the rest of the disk.
+    (
+    echo n
+    echo p
+    echo
+    echo
+    echo
+    echo w) | fdisk "${device}"
+    check_return "fdisk-${device}-end-partition"
+    # sync and wait 5 sec
+    sync
+    sleep 5
+}
+# Usage:
+# format_partition "/dev/sdb" "ext4"
+# shellcheck disable=SC2039
+format_partition() {
```

Comment:
shouldn't it be named 'format_partitions' ? The code suggests it formats all partitions in the device.

Reply:
right.